### PR TITLE
feat: presexch: copy @context and type from doc to BBS+ JSON-LD frame

### LINKED ISSUE #2488: presexch: copy "@context" and "type" from doc to BBS+ JSON-LD frame
When building BBS+ JSON-LD on VP creation at Presentation Definition, we need to copy `@context` and `type` fields from the input VC document if we apply JSON-LD framing on the particular JSON-LD level. Otherwise, we get still a valid JSON-LD frame document but its appliance can lead to an invalid output VP document.

### PR DESCRIPTION
**Pull request by kdimak**
For every field from constraints, do not just set `@explicit` in the reveal doc, but also copy special fields like `type` and `context` from the source VC document to the reveal document if present. Also, we need to copy the parent special fields if present as well.

closes #2488

Signed-off-by: Dmitriy Kinoshenko <dkinoshenko@gmail.com>

### CODE REVIEW

File: pkg/doc/presexch/definition.go [hunk 1]
```diff
@@ -663,25 +663,55 @@ func createNewCredential(constraints *Constraints, src, limitedCred []byte,
...
 
-	doc["@explicit"] = true
-
 	return credential.GenerateBBSSelectiveDisclosure(doc, []byte(uuid.New().String()), opts...)
 }
 
+func enhanceRevealDoc(explicitPaths map[string]bool, limitedCred, vcBytes []byte) ([]byte, error) {
+	var err error
+
+	for path := range explicitPaths {
```
Comment (soluchok) [inline pkg/doc/presexch/definition.go:682]: We are missing the part when the path is few levels deep.
Let's say we have path `subject.user.address`.
The current implementation sets values only for the full path.
e.g 
```
subject.user.address.@explicit
subject.user.address.type
subject.user.address.@context
```

In that case, we are missing everything between.
e.g
```
# root
@explicit
type
@context

# first level
subject.@explicit
subject.@type
subject.@context

# second level
subject.user.@explicit
subject.user.@type
subject.user.@context
```

File: pkg/doc/presexch/definition.go [hunk 2]
```diff
@@ -663,23 +663,79 @@ func createNewCredential(constraints *Constraints, src, limitedCred []byte,
...
 
-	return credential.GenerateBBSSelectiveDisclosure(doc, []byte(uuid.New().String()), opts...)
+		specialFieldValue := gjson.GetBytes(vcBytes, specialFieldPath)
+		if specialFieldValue.Type == gjson.Null {
+			continue
+		}
+
+		limitedCred, err = sjson.SetBytes(limitedCred, specialFieldPath, specialFieldValue.Value())
+		if err != nil {
+			return nil, err
```
Comment (Baha-sk) [inline pkg/doc/presexch/definition.go:734]: can these err types be wrapped?

**Review by Baha-sk — APPROVED** [review]

**Review by soluchok — APPROVED** [review]

File: pkg/doc/presexch/definition.go (hunk 2 above)
Comment (kdimak) [reply pkg/doc/presexch/definition.go:734]: I just followed the error handling pattern in this source file...

File: pkg/doc/presexch/definition.go (hunk 1 above)
Comment (kdimak) [reply pkg/doc/presexch/definition.go]: Fixed!

**Review by fqutishat — APPROVED** [review]

**Pull request merged**